flow/layers/physical_shape_layer.cc

```diff
@@ -66,6 +66,15 @@ void PhysicalShapeLayer::Preroll(PrerollContext* context,
 void PhysicalShapeLayer::UpdateScene(SceneUpdateContext& context) {
   FML_DCHECK(needs_system_composite());
 
+  if (entity_node_ != nullptr) {
+    if (entity_node_->session() && context.top_entity()) {
```

Comment:
Nit: Just add an assertion for this condition and add the child node as normal.

Reply:
Done.